openfl/federated/task/runner_keras.py

```diff
@@ -281,7 +280,12 @@ def _get_weights_names(obj):
         Returns:
             weight_names (list): The weight name list.
         """
-        weight_names = [weight.name for weight in obj.weights]
+        if isinstance(obj, ke.optimizers.Optimizer):
+            weight_names = [weight.name for weight in obj.variables]
+        else:
+            weight_names = [
+                layer.name + "/" + weight.name for layer in obj.layers for weight in layer.weights
+            ]
```

Comment:
Which other optimizers would trigger the `else` block?

Reply:
else is for model weights and if is for Optimizer weights